lib/connectors_async/job_store.rb

```diff
@@ -16,6 +17,8 @@ class JobNotFoundError < StandardError; end
     def initialize
       # multiple threads can write to the store
       @store = Concurrent::Hash.new
+      @watcher = ConnectorsAsync::JobWatcher.new(job_store: self)
```

Comment:
If we make `JobStore` and `JobWatcher` singleton, they don't need to reference each other any more.

Reply:
They still will have dependencies on each other, just implicit.

I'm considering the singleton change, but haven't be able to approach it yet - hopefully soon though.